### ISSUE TITLE
PDF parsing doesn't support partially numbered lists

### ISSUE THREAD

**Issue by majdalsado** [microsoft/markitdown#68]
When parsing a PDF file with partial numberings as is common in [MasterFormat](https://en.wikipedia.org/wiki/MasterFormat), a format used in most construction and government documents across US/Canada, the parser fails to show numbered lists properly. See example.

## MarkItDown Output:
> RFP for Construction Management Services
> Rotary Clubs of Grande Prairie Wellness Centre Society
> Ken Sargent House
> 
> Section 00 00 43
> Instructions to Respondents
> Page 1 of 8
> 
> 1.
> 
> INTENT
> 
> .1
> 
> .2
> 
> .3
> 
> .4
> 
> .5
> 
> .6
> 
> .7
> 
> .8
> 
> The  intent  of  this  Request  for  Proposal  (RFP)  is  to  solicit  submissions,  in  the  format
> detailed in this document, from qualified Construction Managers for the following project:
> 
> KEN SARGENT HOUSE
> GRANDE PRAIRIE, ALBERTA
> 
> Available information relative to the project is included in Section 00 00 45 – Description
> of Project.
> 

## Actual Document
<img width="709" alt="Image" src="https://github.com/user-attachments/assets/bd560965-b1fc-401b-b08d-5bb6e7c41fe3" />

**Comment by afourney** [comment]
Yeah, PDFs are not the nicest format for conversion, and their conversion is perhaps the least good in terms of fidelity. I'd like to improve that, and will investigate some options while keeping things lightweight.
